src/components/Pagination/tests/Pagination.test.tsx

```diff
@@ -78,6 +78,45 @@ describe('<Pagination />', () => {
     });
   });
 
+  describe('accessibilityLabel', () => {
+    it('inserts prop as aria-label', () => {
+      const accessibilityLabel = 'test';
+      const pagination = mountWithAppProvider(
+        <Pagination accessibilityLabel={accessibilityLabel} />,
+      );
+      expect(pagination.find('nav').prop('aria-label')).toStrictEqual(
+        accessibilityLabel,
+      );
+    });
+
+    it('uses default value for aria-label', () => {
+      const pagination = mountWithAppProvider(<Pagination />);
+      expect(pagination.find('nav').prop('aria-label')).toStrictEqual(
+        'Pagination',
+      );
+    });
+  });
+
+  describe('label', () => {
+    it('renders as text', () => {
+      const label = 'test';
+      const pagination = mountWithAppProvider(<Pagination label={label} />);
+      expect(pagination.text()).toContain(label);
+    });
+
+    it('has subdued text without next and previous pages', () => {
+      const label = 'test';
+      const variation = 'subdued';
+      const pagination = mountWithAppProvider(<Pagination label={label} />);
+      expect(
+        pagination
+          .find('.Label')
+          .children()
+          .prop('variation'),
+      ).toStrictEqual(variation);
```

Comment:
Do you need `.find('.Label').children()` over `.find('.TextStyle')`, or was that not working?


Also, is there a reason you put these values in variables? I think the test might read a bit clearer with less indirection, what do you think?

```js
const pagination = mountWithAppProvider(<Pagination label="label" />);
expect(
  pagination
    .find('.TextStyle')
    .prop('variation'),
).toStrictEqual('subdued');
```

Reply:
I tried doing it with .Textstyle and it was running into some issues finding the specific child to compare to. Using my own class from the label component seemed to do the trick but if there's a better way to go about it then I think it's something worth changing.

And I agree that might be a little overkill for the variables if they're only used once, a line after they were defined. I think I just put those in to follow the pattern of creating a variable for the component itself to assert.

Reply:
Changes have been made 🔥